From 78d2239c8cacc2abefd2b45b9015d0370e90b06a Mon Sep 17 00:00:00 2001
From: Michael Rehn <rehn.michael@outlook.com>
Date: Fri, 5 Oct 2018 09:54:58 +0200
Subject: [PATCH 2/2] Fixed patch for group B: fly to any vacant spot

---
 GamePlatform/StdinController.py | 24 ++++++++++++++----------
 1 file changed, 14 insertions(+), 10 deletions(-)

diff --git a/GamePlatform/StdinController.py b/GamePlatform/StdinController.py
index 0238a28..4a362e1 100644
--- a/GamePlatform/StdinController.py
+++ b/GamePlatform/StdinController.py
@@ -69,7 +69,8 @@ class StdinController(Controller):
                 print(
                     "\033[31mYou have to select one of your opponents stones\u001b[0m"
                 )
-            elif len(slots_outside_mills) > 0 and selected_slot not in slots_outside_mills:
+            elif len(slots_outside_mills
+                     ) > 0 and selected_slot not in slots_outside_mills:
                 print("\033[31mYou have to remove stones not in a mill "
                       "before you remove stones inside a mill\u001b[0m")
             else:
@@ -108,13 +109,24 @@ class StdinController(Controller):
 
             if not is_owner:
                 print("This stone is not yours. Select another.")
-            elif not can_move:
+            elif not can_move and not can_fly:
                 print("This stone can't move anywhere, pick another one")
             else:
                 is_valid_input = True
 
         new_position_slot = None
         while new_position_slot is None:
+            if can_fly:
+                chosen_slot = self.get_input_slot(
+                    board, "Your stones can now fly: where do you want to go?")
+                if chosen_slot.owner is None:
+                    new_position_slot = chosen_slot
+                    break
+                else:
+                    print("\033[31mSlot is already occupied, try again\u001b[0m")
+                    continue
+
+
             # Where we want to go
             new_position = input("Choose direction to move: ")
 
@@ -130,14 +142,6 @@ class StdinController(Controller):
                 direction = "bottom"
 
             new_position_slot = getattr(current_position_slot, direction)
-            fly_slot = getattr(new_position_slot, direction)
-
-            if can_fly and fly_slot is not None and fly_slot.owner is None:
-                answer = input("Do you want to move two steps? (yes/no) ")
-                while answer not in ("yes", "no"):
-                    answer = input("Do you want to move two steps? (yes/no) ")
-                if answer == "yes":
-                    new_position_slot = fly_slot
 
             if new_position_slot is None:
                 print("There is no board slot in this direction. Try again!")
-- 
2.19.0.windows.1

